docs/service-fabric/refactor-migrated-app.md

```diff
@@ -0,0 +1,155 @@
+
+![](./images/surveys_02.png)
+
+The **Tailspin.Web** service is ported from the original *Tailspin.Web* web role.
+
+The **Tailspin.Web.Survey.Public** service is ported from the original *Tailspin.Web.Survey.Public* web role.
+
+The **Tailspin.AnswerAnalysisService** service is ported from the original *Tailspin.Workers.Survey* worker role.
+
+> [!NOTE] 
+> While minimal code changes were made to each of the web and worker roles, **Tailspin.Web** and **Tailspin.Web.Survey.Public** were modified to self-host a Kestrel web server. 
+
+Now, Tailspin is refactoring the Surveys application to a more granular architecture. Tailspin's motivation for refactoring is to make it easier to develop, build, and deploy the Surveys application. By decomposing the existing web and worker roles to a more granular architecture, Tailspin wants to remove the existing tightly coupled communication and data dependencies between these roles.
+
+Tailspin sees other benefits in moving the Surveys application to a more granular architecture:
+* Each service can be packaged into independent projects with a scope small enough to be managed by a small team.
+* Each service can be independently versioned and deployed.
+* Each service can be implemented using the best technology for that service.
+
+For example, a service fabric cluster can include services built using different versions of the .Net Frameworks, Java, or other languages such as C or C++.
+
+> [!NOTE] 
+> Multitenancy is out of scope for the refactoring of this application. Tailspin has several options to support multitenancy and can make these design decisions later without affecting the initial design. For example, Tailspin can create separate instances of the services for each tenant within a cluster or create a separate cluster for each tenant.
+
+## Design considerations
+
+The following diagram shows the architecture of the Surveys application refactored to a more granular architecture:
+
+![](./images/surveys_03.png)
+
+**Tailspin.Web** is a stateless service self-hosting an ASP.NET MVC application that Tailspin customers visit to create surveys and view survey results. This service shares most of its code with the *Tailspin.Web* service from the ported Service Fabric application. As mentioned earlier, this service uses ASP.NET core and switches from using Kestrel as web frontend to implementing a WebListener.
+
+**Tailspin.Web.Surveys.Public** is a stateless service also self-hosting an ASP.NET MVC site. Users visit this site to select surveys from a list and then fill them out. This service shares most of its code with the *Tailspin.Web.Survey.Public* service from the ported Service Fabric application. This service also uses ASP.NET Core and also switches from using Kestrel as web frontend to implementing a WebListener.
```

Comment:
Why did it switch to WebListener?

Reply:
The switch happened in the VS templates for SF services. Kestral -> WebListener